activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -789,6 +803,25 @@ def construct_coder(row, coder_class)
           coder_class.new(oid: row['oid'].to_i, name: row['typname'])
         end
 
+        # PostgreSQL requires two 32bit integers as the lock keys
+        # PostgreSQL advisory locks are scoped to a database by default so no
+        # need to add a database name
+        def transform_lock_key(key_string, prefix = nil)
+          stable_hashcode =
+            if key_string.is_a?(Numeric)
+              key_string.to_i
+            else
+              # Ruby MRI's String#hash is randomly seeded as of Ruby 1.9 so
+              # make sure we use a deterministic hash.
+              Zlib.crc32(key_string.to_s)
+            end
+
+          [stable_hashcode, prefix].map do |ea|
+            # pg advisory args must be 31 bit ints
+            ea.to_i & 0x7fffffff
+          end
+        end
```

Comment:
Sure, we could use a pre-defined integer. Two reasons why I didn't:

1. We need to mangle for MySQL anyway because advisory locks are server-wide and not scoped to a database
2. If we implement it this way we get generic advisory lock support on the connection in all of our Rails apps for free.